source-boilerplate/boilerplate.go

```diff
@@ -25,11 +26,24 @@ func RunMain(srv pc.DriverServer) {
 	var parser = flags.NewParser(nil, flags.Default)
 	var ctx, _ = signal.NotifyContext(context.Background(), syscall.SIGTERM, syscall.SIGINT)
 
+	var conn net.Conn
+	if server, err := net.Listen("tcp", ":2222"); err != nil {
+		logrus.WithFields(logrus.Fields{"error": err}).
+			Error("failed to start tcp server")
```

Comment:
You can use `logrus.Fatal()` here, it's basically just a helper for `logrus.Error(); os.Exit(1)` already.